pkgs/development/ocaml-modules/reason-native/default.nix

```diff
@@ -0,0 +1,183 @@
+{ lib, fetchFromGitHub, buildDunePackage, ocamlPackages }:
```

Comment:
Since this is part of `ocamlPackages`, you should take each OCaml dependency as a separate argument (instead of the whole `ocamlPackages` set).

Reply:
As I am using pkgs.callPackage`s` (emphasis on `s`) on toplevel. I have to manually add dependencies at toplevel, that might not be a good idea, but I did not find a way of workaround this.
I could pass ocamlPackages and inherit at reason-native from ocamlPackages, which way you consider best?
I have proposed the version that minimally lists required packages.

Reply:
As pkgs.callPackages (emphasis on `s`) is used at toplevel. On my experiments, OCaml dependencies would either have to be added at toplevel adding noise (undesirable) or be inherited from ocamlPackages at reason-native.

Updated code to inherit from ocamlPackages at reason-native. Like:
```
with {
  inherit (ocamlPackages) buildDunePackage reason re pastel qcheck-core atdgen junit; 
};
```
Seems fine.